Library/brew.sh

```diff
@@ -196,39 +197,8 @@ EOS
   esac
 fi
 
-if [[ -n "$HOMEBREW_ANALYTICS" ]]
-then
-  # User UUID file. Used for Homebrew user counting. Can be deleted and
-  # recreated with no adverse effect (beyond our user counts being inflated).
-  HOMEBREW_ANALYTICS_USER_UUID_FILE="$HOME/.homebrew_analytics_user_uuid"
-  if [[ -r "$HOMEBREW_ANALYTICS_USER_UUID_FILE" ]]
-  then
-    HOMEBREW_ANALYTICS_USER_UUID="$(<"$HOMEBREW_ANALYTICS_USER_UUID_FILE")"
-  else
-    HOMEBREW_ANALYTICS_USER_UUID="$(uuidgen)"
-    echo "$HOMEBREW_ANALYTICS_USER_UUID" > "$HOMEBREW_ANALYTICS_USER_UUID_FILE"
-  fi
-  export HOMEBREW_ANALYTICS_ID="UA-75654628-1"
-  export HOMEBREW_ANALYTICS_USER_UUID
-
-  # Send the to-be-executed command as an "App Screen View". Anonymise the IP
-  # address (aip=1) and don't send or store any personally identifiable
-  # information.
-  # https://developers.google.com/analytics/devguides/collection/protocol/v1/devguide#screenView
-  # https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters
-  "$HOMEBREW_CURL" https://www.google-analytics.com/collect -d v=1 \
-    --silent --max-time 3 --output /dev/null \
-    --user-agent "$HOMEBREW_USER_AGENT_CURL" \
-    -d tid="$HOMEBREW_ANALYTICS_ID" \
-    -d cid="$HOMEBREW_ANALYTICS_USER_UUID" \
-    -d aip=1 \
-    -d an=Homebrew \
-    -d av="$HOMEBREW_VERSION" \
-    -d t=screenview \
-    -d cd="$HOMEBREW_COMMAND" \
-    &> /dev/null \
-    & disown
-fi
+source "$HOMEBREW_LIBRARY/Homebrew/utils/analytics.sh"
+report-analytics-screenview-command
```

Comment:
Feels weird to source a file and have that executed a function (we don't do that elsewhere).